<!DOCTYPE HTML>
<html>

<script src="js/scripts.js"></script>
<script src="js/gcse.js"></script>
<script src="js/lightbox-plus-jquery.js"></script>

<head>
  <title>Newt Editor</title>
  <meta name="description" content="Newt Pathway Viewer & Editor" />
  <meta name="keywords" content="Pathway, Biological Networks, Visualization, Editor, Curator, SBGN, Systems Biology Graphical Notation" />
  <meta http-equiv="content-type" content="text/html; charset=windows-1252" />
  <link rel="icon" href="style/newt-icon-blue.png">
  <link href="style/lightbox.css" rel="stylesheet">
  <link rel="stylesheet" type="text/css" href="style/style.css" title="style" />
</head>

<body>
  <div id="main">
    <div id="header">
      <script> writeLogo(); </script>
      <script> writeMenubar(4); </script>
    </div>
    <div id="site_content">
      <div class="sidebar">
      <script> writeNews(); </script>
      </div>
      <div id="content">
        <h1 id="top-of-tutorials">Tutorials</h1>

        <ul>
        <li> <a href="#introduction">Introduction</a> 
             <span style="float: right;">Newt basics video: <a href="tutorials-basics.html"><img src="images/play-button.svg" width=16px/></a></span>
        <li> <a href="#menubar-and-toolbar">Menubar and toolbar</a>
        <li> <a href="#right-panel">Right panel</a>
        <li> <a href="#file-operations">File operations</a>
        <li> <a href="#convenient-editing">Convenient editing</a>
        <li> <a href="#alignment-grid-and-guidelines">Alignment, grid, and guidelines</a>
        <li> <a href="#complexity-management">Complexity management</a>
             <span style="float: right;">Complexity management video: <a href="tutorials-compl.html"><img src="images/play-button.svg" width=16px/></a></span>
        <li> <a href="#search-and-highlight">Search and highlight</a>
        <li> <a href="#automatic-layout">Automatic layout</a>
             <span style="float: right;">Automatic layout video: <a href="tutorials-layout.html"><img src="images/play-button.svg" width=16px/></a></span>
        <li> <a href="#querying-pathway-commons">Querying Pathway Commons</a>
        <li> <a href="#references">References</a>
        </ul>

        <h2 id="introduction">Introduction
          <a href="#top-of-tutorials"><img src="images/back-to-top.svg" height=16px/></a></h2>

        <p> Newt editor is composed of a menu bar and a toolbar laid out horizontally on top of the drawing canvas and a right panel for object inspection, map configuration and the palette. </p>

        <span class="center"><a href="images/newt-ss.png" data-lightbox="image-intro" data-title="Overall look of the Newt editor"><img src="images/newt-ss.png" style="width: 593px;" /></a></span>

        <p> Newt is a viewer and editor of maps drawn in SBGN notation. Specifically, currently it supports Process Description (PD) and Activity Flow (AF) languages of this notation. Please refer to related <a href="http://sbgn.github.io/sbgn/templates">reference cards</a> and <a href="http://sbgn.github.io/sbgn/specifications">specifications</a> for details of these languages. </p>

        <p> <a href="tutorials-basics.html">Here</a> is a short tutorial illustrating basics of Newt. </p>

        <h2 id="menubar-and-toolbar">Menubar and toolbar
          <a href="#top-of-tutorials"><img src="images/back-to-top.svg" height=16px/></a></h2>

        <p> Menubar located at the top of the Newt editor is where all capabilities of the tool is assembled in an organized manner. Right below is a toolbar, which compiles most frequently used of all available operations for convenience. </p>

        <h2 id="right-panel">Right panel
          <a href="#top-of-tutorials"><img src="images/back-to-top.svg" height=16px/></a></h2>

        <p> The panel on the right is composed of the following tabs. </p>

        <h3>Inspector</h3>
        <span class="left"><a href="images/inspector.png" data-lightbox="image-right-panel" data-title="Editing through object inspector"><img src="images/inspector.png" style="width: 400px;"/></a></span>
        <p> The object inspector on the right panel titled "Object" is activated when a particular object is selected by left-clicking on it, enabling inspection and editing of the style of the underlying object from border color to font type and size to state variables and custom annotations.</p>
        <p> When multiple objects are selected, inspector only display the common attributes (and their values if all objects have the same value for that attribute) for convenient modification of those attributes. </p>

        <h3>Map settings</h3>

        <span class="right"><a href="images/map-properties.png" data-lightbox="image-right-panel" data-title="Configuring the map through options"><img src="images/map-properties.png" style="width: 400px;"/></a></span>
        <p> The tab titled "Map" on the right panel is for configuring the drawing of the map. It can be enabled with "File | Map Properties..." or through the associated tool on the toolbar. </p>
        <p> Numerous settings of a map from general properties like compound paddings to label related settings such as whether to fit labels to nodes, to color scheme can be set using this tab. </p>

        <h3>Palette</h3>
        <span class="left"><a href="images/palette.png" data-lightbox="image-right-panel" data-title="Constructing map using the palette"><img src="images/palette.png" style="width: 400px;"/></a></span>
        <p> When Palette tab on the right panel is chosen or when the user clicks in an empty area on the drawing canvas, the palette composed of PD and AF nodes and edges, is displayed. The type of a map (see Type under General Properties) is considered PD (AF) when the user constructs a map using only nodes and edges from the PD (AF) palette. When the user mixes up nodes and/or edges from both palettes though, its type is set as "Unknown", and notation specific checks (e.g. a modulation arc can only go from a source EPN such as macromolecule to a process node) are no longer performed. </p>

        <p> Nodes can either be created by choosing and dragging from the palette onto the canvas or by selecting a particular node on the palette and then clicking on the canvas. Clicking on a node or an edge puts you in node/edge creation mode, and upon creation of the node/edge, Newt returns to selection mode. Clicking twice on a tool, on the other hand, puts you in sustained mode so that the user can create multiple objects of that type in a row. </p>

        <h2 id="file-operations">File operations
          <a href="#top-of-tutorials"><img src="images/back-to-top.svg" height=16px/></a></h2>
        <p> Newt can read and write an extended version of <a href="https://github.com/sbgn/sbgn/wiki/SBGN_ML">SBGNML</a> files. Extension includes style of map objects (such as color and font), custom properties (e.g. Uniprot ID associated with a macromolecule), and hidden/collapsed map content; thus, such information is restored upon save-reload. </p>

        <p> In addition, File menu can be used to export the current map as a static image (either as a raster file in PNG or JPG or as scalable vector graphics in SVG).</p>

        <h2 id="convenient-editing">Convenient editing
          <a href="#top-of-tutorials"><img src="images/back-to-top.svg" height=16px/></a></h2>

        <p> Map content can be expanded using the Palette as described earlier. A convenient way to populate a map is using the templates under the Edit menu. </p>

        <table class="plaintable">
          <tr>
            <th class="leftcolumn">
            <a href="images/template-dialog.png" data-lightbox="image-templates" data-title="Template dialog for convenient editing"><img class="ssthumbnail" src="images/template-dialog.png" style="height: 220px;"></a>
            </th>
            <th class="rightcolumn">
            <a href="images/template-association-example.png" data-lightbox="image-templates" data-title="Example creation of an association using the template dialog"><img class="ssthumbnail" src="images/template-association-example.png" style="height: 100px;"></a>
            </th> 
          </tr>
        </table>

        <h2 id="alignment-grid-and-guidelines">Alignment, grid and guidelines
          <a href="#top-of-tutorials"><img src="images/back-to-top.svg" height=16px/></a></h2>

        <p> Newt provides various mechanisms for alignment of map objects. One such facility is available through the View menu, where the user can select a number of objects and align their center or edge horizontally or vertically. Such alignment is performed with respect to the object selected first. </p>

        <p> Another way to perform alignment is through the grid. When you enable the grid tool (<img src="images/enable-grid.svg" title="Tool for enabling the grid" style="height: 16px;">) on the toolbar, a grid is displayed and nodes are snapped to grid as they are dragged. </p>

        <span class="right"><a href="images/guideline-example.png" data-lightbox="image-grid-guide" data-title="An example where the dragged node is snapped to an alignment position"><img src="images/guideline-example.png" style="width: 340px;"/></a></span>
        <p> Yet another way to perform alignment is by enabling the guidelines with the associated tool on the toolbar (<img src="images/enable-guidelines.svg" title="Tool for enabling the guidelines" style="height: 16px;">). Two types of guidelines are enabled with this tool:
<ul>
 <li> Geometric guidelines (e.g. center of two nodes are in alignment) and
 <li> Distance guidelines (e.g. a process node is distanced equally from its source and target).
</ul>
Grid and guideline support is rather involved and can be customized using the settings through the View menu.
</p>

        <h2 id="complexity-management">Complexity management
          <a href="#top-of-tutorials"><img src="images/back-to-top.svg" height=16px/></a></h2>

        <p> Some maps can be rather large. No matter how advanced of layout and viewing facilities are available, navigation through such maps and tracing interactions can be cumbersome. Newt provides some advanced complexity management techniques [<a href="#references">2</a>,<a href="#references">1</a>,<a href="#references">3</a>] so that the user can focus on a sub-map of interest at any time during their analysis, removing unwanted details <i>temporarily</i> from the map. Below we discuss these facilities. </p>

        <p> <a href="tutorials-compl.html">Here</a> is a short tutorial illustrating complexity management in Newt. </p>

        <h3>Hide-show</h3>
        <p> Unwanted details of a map may be hidden by the user temporarily through the View menu. Below is an example, where the user initially keeps a single process and hides all others (top). Then the user gradually reveals the hidden neighbors (bottom maps) using "Show Hidden Neighbors" in the node context menu. </p>

        <table class="plaintable">
          <tr>
            <th colspan="4">
            <a href="images/hide-example-1.png" data-lightbox="image-hide" data-title="Map content is hidden except for a process"><img class="ssthumbnail" src="images/hide-example-1.png" style="height: 360px;"/></a>
            </th>
          </tr>
          <tr>
            <th>
            <a href="images/hide-example-2.png" data-lightbox="image-hide" data-title="Then the user starts to gradually show hidden content"><img class="ssthumbnail" src="images/hide-example-2.png" style="height: 70px;"/></a>
            </th>
            <th>
            <a href="images/hide-example-3.png" data-lightbox="image-hide" data-title="One more level of neighborhood is revealed"><img class="ssthumbnail" src="images/hide-example-3.png" style="height: 86px;"/></a>
            </th>
            <th>
            <a href="images/hide-example-4.png" data-lightbox="image-hide" data-title="Unhide operation continues as deep as desired"><img class="ssthumbnail" src="images/hide-example-4.png" style="height: 100px;"/></a>
            </th>
            <td>...</td>
          </tr>
        </table>

        <h3>Expand-collapse</h3>
        <p> Another way to manage complexity is by <i>collapsing</i> currently not-so-interesting parts of a map (i.e. a submap, a compartment or a complex). Collapsed compound node can later be <i>expanded</i> to reveal its nested content. </p>

        <p> Below is an example, where, first, all complexes of a map have been collapsed to reduce complexity. Then, nucleus was collapsed to focus on the rest of the map. </p>
        <table class="plaintable">
          <tr>
            <th colspan="2">
            <a href="images/expand-collapse-example-1.png" data-lightbox="image-expand" data-title="Original graph is composed of many complexes with members shown, and a couple of cellular compartments"><img class="ssthumbnail" src="images/expand-collapse-example-1.png" style="height: 240px;"/></a>
            </th>
          </tr>
          <tr>
            <th>
            <a href="images/expand-collapse-example-2.png" data-lightbox="image-expand" data-title="After all complexes are collapsed"><img class="ssthumbnail" src="images/expand-collapse-example-2.png" style="height: 220px;"/></a>
            </th>
            <th>
            <a href="images/expand-collapse-example-3.png" data-lightbox="image-expand" data-title="After nucleus is collapsed"><img class="ssthumbnail" src="images/expand-collapse-example-3.png" style="height: 132px;"/></a>
            </th>
          </tr>
        </table>

        <h2 id="search-and-highlight">Search and highlight
          <a href="#top-of-tutorials"><img src="images/back-to-top.svg" height=16px/></a></h2>
        <p> Newt allows users to highlight a specified part of a map through the Highlight menu (Highlight | Selected). To highlight the part of a map involving a certain object, you may use the Search facility (Highlight | Search). Newt looks for node labels with specified keyword and highlights them (all processes involved in case of map type being PD). </p>
        <span class="center"><a href="images/search-example.png" data-lightbox="image-search" data-title="Part of the map containing &quot;fructose&quot; in its label is searched for and highlighted"><img src="images/search-example.png" style="width: 500px;" /></a></span>

        <h2 id="automatic-layout">Automatic layout
          <a href="#top-of-tutorials"><img src="images/back-to-top.svg" height=16px/></a></h2>

        <span class="left"><a href="images/before-layout.svg" data-lightbox="image-layout" data-title="A map with random layout"><img src="images/before-layout.svg" style="height: 220px;" /></a></span>
        <span class="right"><a href="images/after-layout.svg" data-lightbox="image-layout" data-title="After Newt automatic layout performed on the same map"><img src="images/after-layout.svg" style="height: 220px;" /></a></span>

        <p> Newt provides automatic layout services to lay out a map from scratch (disable Incremental flag under Layout | Layout Properties...) or incrementally (default). Newt uses a layout algorithm [<a href="#references">5</a>,<a href="#references">4</a>,<a href="#references">6</a>] respecting compound structures (e.g. compartments). </p>

        <p> The same algorithm is also applied incrementally after complexity management operations described earlier, if the associated flag (Map Properties... | Arrangement | Rearrange on Complexity Management) is enabled. </p>

        <p> <a href="tutorials-layout.html">Here</a> is a short tutorial illustrating automatic layout capabilities of Newt. </p>

        <h2 id="querying-pathway-commons">Querying Pathway Commons
          <a href="#top-of-tutorials"><img src="images/back-to-top.svg" height=16px/></a></h2>

        <p> Newt provides an interface to perform live queries to the <a target="_blank" href="http://pathwaycommons.org/">Pathway Commons</a> database, presenting results in SBGN, and enabling editing on them. </p>

        <span class="right"><a href="images/PC-query-result.png" data-lightbox="image-layout" data-title="Partial view of the result of sample query"><img src="images/PC-query-result.png" style="height: 240px;" /></a></span>
        <span class="right"><a href="images/PC-query-dialog.png" data-lightbox="image-layout" data-title="Sample Paths Between query"><img src="images/PC-query-dialog.png" style="height: 140px;" /></a></span>
        <p> Currently available type of queries include Query by URI and Query Paths Between (find the paths between specific source set of states or entities within the boundaries of a specified length limit) as detailed <a target="_blank" href="http://www.pathwaycommons.org/pc2/">here</a> [<a href="#references">7</a>]. Paths Between query takes a list of gene symbols (separated by blank spaces or on separate lines) and a length limit specifying the maximum length of paths between specified genes. Since there could be exponentially many paths between multiple genes, this length limit can be at most 3.</p>

        <h2 id="references">References
          <a href="#top-of-tutorials"><img src="images/back-to-top.svg" height=16px/></a></h2>

        [1] M. Sari, I. Bahceci, U. Dogrusoz, S.O. Sumer, B.A. Aksoy, O. Babur, E. Demir, "SBGNViz: a tool for visualization and complexity management of SBGN process description maps", PLoS ONE, 10(6), e0128985, 2015. 
        <br><br>[2] U. Dogrusoz and B. Genc, "A Multi-Graph Approach to Complexity Management in Interactive Graph Visualization", Computers & Graphics, vol. 30/1, pp. 86-97, 2006.
        <br><br>[3] U. Dogrusoz and B. Genc, "A Framework for Complexity Management in Graph Visualization", LNCS, vol. 2528, pp. 368-369, 2002.
        <br><br>[4] B. Genc, U. Dogrusoz, "An Algorithm for Automated Layout of Process Description Maps Drawn in SBGN", Bioinformatics, 32(1), pp. 77-84, 2016.
        <br><br>[5] U. Dogrusoz, E. Giral, A. Cetintas, A. Civril, and E. Demir, "A Layout Algorithm For Undirected Compound Graphs", Information Sciences, 179, pp. 980-994, 2009.
        <br><br>[6] U. Dogrusoz, E. Giral, A. Cetintas, A. Civril, and E. Demir, "A Compound Graph Layout Algorithm for Biological Pathways", LNCS, vol. 3383, pp. 442-447, 2004.
        <br><br>[7] U. Dogrusoz, A. Cetintas, E. Demir, and O. Babur, "Algorithms for Effective Querying of Compound Graph-Based Pathway Databases", BMC Bioinformatics, 10:376, 2009. 
 
      </div>
    </div>
    <script> writeFooter(); </script>
  </div>
</body>
</html>
